screen/contributors/src/main/java/com/ivy/contributors/ContributorsViewModel.kt

```diff
@@ -49,22 +49,28 @@ class ContributorsViewModel @Inject constructor(
     }
 
     private suspend fun fetchContributors() {
-        val contributors = ivyWalletRepositoryDataSource.fetchContributors()?.map {
-            Contributor(
-                name = it.login,
-                photoUrl = it.avatarUrl,
-                contributionsCount = it.contributions.toString(),
-                githubProfileUrl = it.link
-            )
-        }
+        val contributorsResult = ivyWalletRepositoryDataSource.fetchContributors()
 
-        if (contributors != null) {
-            contributorsResponse.value = ContributorsResponse.Success(
-                contributors.toImmutableList()
-            )
-        } else {
-            contributorsResponse.value = ContributorsResponse.Error("Error")
-        }
+        contributorsResult.fold(
```

Comment:
contributorsResponse.value = contributorsResult.fold(...)

Reply:
Got it. We should use Either on the screen as well, right?

Reply:
Read this:
https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/guidelines/Data-Modeling.md

Use ADTs, sealed interface with one for Loading, Success, Error